### PR TITLE
Add svc-bgs-api integration tests as a Gradle task

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?

Integration tests were implemented within our GitHub actions workflow which prevented them from being run locally and also created undue complexity.

Associated tickets or Slack threads:
- #2172 

## How does this fix it?[^1]

Converts svc-bgs-api integration tests to 

## How to test this PR
- Spin up the platform docker containers along with the BGS container and run `./gradlew :svc-bgs-api:integrationTest`
- Check for successful run of the `svc-bgs-api` integration test GH workflow on this PR

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Comment by msnwatson** [comment]
Failing checks are not caused by this PR

File: svc-bgs-api/src/integrationTest/add_note_tests.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+require_relative '../config/constants'
+
+# Constants used by integration tests
+ADD_NOTE_REPLY_QUEUE = "add-note-response"
+ADD_NOTE_PAYLOAD = "{\\\"veteranNote\\\":\\\"test\\\",\\\"veteranParticipantId\\\":111}"
+CORRELATION_ID = "1234"
+
+
+def test_successful_add_note(ch)
+    x = ch.direct(BGS_EXCHANGE_NAME, CAMEL_MQ_PROPERTIES)
```
Comment (tejans24) [inline svc-bgs-api/src/integrationTest/add_note_tests.rb:15]: I suggest we add more descriptive names for variables even in tests

**Review by Ponnia-M — APPROVED** [review]

File: svc-bgs-api/src/integrationTest/add_note_tests.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+require_relative '../config/constants'
+
+# Constants used by integration tests
+ADD_NOTE_REPLY_QUEUE = "add-note-response"
+ADD_NOTE_PAYLOAD = "{\\\"veteranNote\\\":\\\"test\\\",\\\"veteranParticipantId\\\":111}"
+CORRELATION_ID = "1234"
+
+
+def test_successful_add_note(ch)
```
Comment (tejans24) [inline svc-bgs-api/src/integrationTest/add_note_tests.rb:14]: We should plan on using a proper testing lib in ruby such as RSpec // Minitest etc

**Review by tejans24 — COMMENTED** [review]
Adding a couple of comments here

File: svc-bgs-api/src/integrationTest/add_note_tests.rb (hunk 2 above)
Comment (msnwatson) [reply svc-bgs-api/src/integrationTest/add_note_tests.rb:14]: yeah it looks like right now, we'd have to change our application model to be more module based in order to really leverage these tools, and I didn't want to make those changes as part of just getting our integration tests out of GH actions.

**Review by agile-josiah — APPROVED** [review]
LGTM

**Pull request merged**